[Wave 200pts] Fix firebase.service.spec.ts TypeScript errors and broken NestJS test module setup — get test suite to zero failures

## Summary
`jest_error.json` committed in the repo root records a live test failure on a developer's machine. The test file `src/firebase/firebase.service.spec.ts` fails with two TypeScript errors:
1. `TS2307: Cannot find module 'firebase-admin'` — the module is now in `package.json` but the spec file is not correctly configured to resolve it in the Jest environment
2. `TS2339: Property 'get' does not exist on type 'TestingModule'` — the test uses an outdated NestJS Testing API pattern

Both errors crash the entire test suite before any test runs. `npm test` exits non-zero for every contributor who clones the repo.

## Why This Matters
A broken test suite means CI cannot be enabled. This is the single blocking issue preventing automated testing from being enforced on any PR. It is listed as an active failure in committed evidence in the repo.

## What Needs to Be Done
- Fix `src/firebase/firebase.service.spec.ts`:
  - Add a Jest module name mapper for `firebase-admin` or create a manual mock at `src/__mocks__/firebase-admin.ts`
  - Fix the `module.get<FirebaseService>` API — the correct call is `moduleRef.get(FirebaseService)` where `moduleRef` is the result of `Test.createTestingModule(...).compile()`
  - Mock `admin.initializeApp()` and `admin.messaging()` so no real Firebase connection is attempted
- Write 3 meaningful unit tests for `FirebaseService`: (a) initializes without throwing, (b) `sendToDevices()` with valid tokens calls messaging send, (c) `sendToDevices()` with an expired/invalid token removes it from the user record
- Run `npm test` — confirm exit code 0 and zero failing suites
- Remove `jest_error.json`, `jest-out.txt`, `tsc_output.txt`, `tsc_final.txt` from the repo with `git rm --cached` and add to `.gitignore`

## Key Files
- `src/firebase/firebase.service.spec.ts` — fix TypeScript errors and test setup
- `src/__mocks__/firebase-admin.ts` — create manual Jest mock (optional approach)
- `.gitignore` — add the four artifact filenames

## Acceptance Criteria
- [ ] `npm test` exits with code 0
- [ ] `npx tsc --noEmit` exits with zero errors
- [ ] `firebase.service.spec.ts` has 3 passing unit tests
- [ ] No real Firebase API is called during test runs
- [ ] `jest_error.json`, `jest-out.txt`, `tsc_output.txt`, `tsc_final.txt` removed from repo root and added to `.gitignore`

## Constraints
- Do not call real Firebase APIs in any test — always mock
- Do not change the FirebaseService business logic — only fix the test setup
- Complexity: **High — 200 points**

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -57,6 +57,12 @@ jobs:
       - name: Type Check
         run: npx tsc --noEmit
 
+      - name: Run TypeORM migrations (ensure DB schema is up to date)
+        env:
+          NODE_ENV: test
+          DATABASE_URL: ${{ secrets.DATABASE_URL || 'postgresql://postgres:password@localhost:5432/nexafx_test' }}
+        run: npm run typeorm:migration:run
+
       - name: Run Unit Tests
         env:
           NODE_ENV: test
@@ -75,6 +81,12 @@ jobs:
           MAILGUN_FROM_EMAIL: 'test@nexafx.com'
           FRONTEND_URL: 'http://localhost:3001'
         run: npm run test -- --forceExit
+
+      - name: Check for migration drift (fail if entity changes are not migrated)
+        env:
+          NODE_ENV: test
+          DATABASE_URL: ${{ secrets.DATABASE_URL || 'postgresql://postgres:password@localhost:5432/nexafx_test' }}
+        run: npm run typeorm:migration:generate -- --check
 
       - name: Build
         run: npm run build

--- a/src/app.module.ts
+++ b/src/app.module.ts
@@ -43,7 +43,12 @@ import { WalletsModule } from './wallets/wallets.module';
       useFactory: (configService: ConfigService) => ({
         type: 'postgres',
         url: configService.get<string>('DATABASE_URL'),
-        synchronize: true,
+        synchronize:
+          process.env.NODE_ENV !== 'production' &&
+          process.env.NODE_ENV !== 'staging',
+        migrationsRun:
+          process.env.NODE_ENV === 'production' ||
+          process.env.NODE_ENV === 'staging',
         ssl:
           process.env.NODE_ENV === 'production'
             ? { rejectUnauthorized: false }

--- a/src/database/data-source.ts
+++ b/src/database/data-source.ts
@@ -3,10 +3,12 @@ import { config } from 'dotenv';
 
 config();
 
+// TypeORM CLI datasource for migrations and CLI commands
+// synchronize: false and correct migrations config are set for safety
 export const AppDataSource = new DataSource({
   type: 'postgres',
   url: process.env.DATABASE_URL,
-  synchronize: false,
+  synchronize: false, // Never auto-sync in CLI
   logging: false,
   entities: ['src/**/*.entity.ts'],
   migrations: ['src/migrations/*.ts'],

